### PR TITLE
fixed subscription names + upgraded provider versions

### PR DESCRIPTION
**Pull request by antonioT90**
### Motivation and context

fixed subscription names + upgraded provider versions

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/core/env/uat/backend.tfvars [hunk 1]
```diff
@@ -1,4 +1,4 @@
 resource_group_name  = "io-infra-rg"
-storage_account_name = "pagopainfraterraformuat"
+storage_account_name = "selcustinfraterraform"
 container_name       = "azureadstate"
```
Comment (pasqualedevita) [inline src/core/env/uat/backend.tfvars:3]: Should be **azurerm** for all

File: src/core/env/prod/backend.tfvars [hunk 2]
```diff
@@ -1,4 +1,4 @@
 resource_group_name  = "io-infra-rg"
-storage_account_name = "pagopainfraterraformprod"
+storage_account_name = "selcpstinfraterraform"
 container_name       = "azureadstate"
 key                  = "prod.terraform.tfstate"
```
Comment (pasqualedevita) [inline src/core/env/prod/backend.tfvars:4]: Should be **terraform.tfstate** for all (we have different container and storage account)

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
Only few changes

File: src/core/env/prod/backend.tfvars (hunk 2 above)
Comment (antonioT90) [reply src/core/env/prod/backend.tfvars:4]: Actual resource name used for UAT is selcustinfraterraform, as created during authorizing process.
Should we first try to manually rename it inside the subscription?

File: src/core/env/uat/backend.tfvars (hunk 1 above)
Comment (antonioT90) [reply src/core/env/uat/backend.tfvars:3]: Actual resource name used for UAT is selcustinfraterraform, as created during authorizing process.
Should we first try to manually rename it inside the subscription?

Comment (pasqualedevita) [reply src/core/env/uat/backend.tfvars:3]: Sorry container name

File: src/core/env/prod/backend.tfvars (hunk 2 above)
Comment (pasqualedevita) [reply src/core/env/prod/backend.tfvars:4]: Sorry it's for key

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
Check also ref to pagopa into vars
-tags
-dns
-apim

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
in `terraform.tfvars` there are other refs to pagopa product

- https://github.com/pagopa/selfcare-infra/blob/upgrade-provider-version/src/core/env/dev/terraform.tfvars#L4
- https://github.com/pagopa/selfcare-infra/blob/upgrade-provider-version/src/core/env/dev/terraform.tfvars#L40
- https://github.com/pagopa/selfcare-infra/blob/upgrade-provider-version/src/core/env/dev/terraform.tfvars#L32

dev.platform -> dev.selfcare
uat.platform -> uat.selfcare
platform -> selfcare

File: src/core/env/dev/terraform.tfvars [hunk 3]
```diff
@@ -5,7 +5,7 @@ tags = {
   CreatedBy   = "Terraform"
   Environment = "Dev"
   Owner       = "pagoPA"
```
Comment (pasqualedevita) [inline src/core/env/dev/terraform.tfvars:7]:
```suggestion
  Owner       = "SelfCare"
```

File: src/core/env/prod/terraform.tfvars [hunk 4]
```diff
@@ -26,22 +26,22 @@ cidr_vnet_integration = ["10.230.7.0/24"] # todo ask to SIA
 cidr_subnet_apim      = ["10.230.7.0/26"]
 
 # dns
-external_domain = "pagopa.it"
-dns_zone_prefix = "platform"
+external_domain = "pagopa.it" # FIXME da chiedere
```
Comment (pasqualedevita) [inline src/core/env/prod/terraform.tfvars:29]: it's ok
```suggestion
external_domain = "pagopa.it"
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — APPROVED** [review]
Top @antonioT90!!

**Comment by pasqualedevita** [comment]
@antonioT90 u can now run the first plan/apply

```
bash terraform.sh plan dev -target=azurerm_resource_group.default_roleassignment_rg
bash terraform.sh apply dev -target=azurerm_resource_group.default_roleassignment_rg
```

**Comment by antonioT90** [comment]
thanks! apply performed with the target provided|

**Pull request merged**